### PR TITLE
Partial integration of the Status Handler

### PR DESCRIPTION
**Pull request by iulianpascalau**
- partial integration of the status handler in bridge
- created eth wrapper

### CODE REVIEW

File: clients/ethereum/wrappers/ethereumChainWrapper.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	return wrapper.multiSigContract.ExecuteTransfer(opts, tokens, recipients, amounts, nonces, batchNonce, signatures)
+}
+
+// Quorum returns the current set quorum value
+func (wrapper *ethereumChainWrapper) Quorum(ctx context.Context) (*big.Int, error) {
+	wrapper.AddIntMetric(core.MetricNumEthClientRequests, 1)
+	return wrapper.multiSigContract.Quorum(&bind.CallOpts{Context: ctx})
+}
+
+func (wrapper *ethereumChainWrapper) GetStatusesAfterExecution(ctx context.Context, batchID *big.Int) ([]byte, error) {
```
Comment (sstanculeanu) [inline clients/ethereum/wrappers/ethereumChainWrapper.go:111]: missing comment

File: clients/ethereum/wrappers/interface.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+type genericErc20Contract interface {
+	BalanceOf(opts *bind.CallOpts, account common.Address) (*big.Int, error)
+}
+
+type multiSigContract interface {
+	GetBatch(opts *bind.CallOpts, batchNonce *big.Int) (contract.Batch, error)
+	GetRelayers(opts *bind.CallOpts) ([]common.Address, error)
+	WasBatchExecuted(opts *bind.CallOpts, batchNonce *big.Int) (bool, error)
+	ExecuteTransfer(opts *bind.TransactOpts, tokens []common.Address, recipients []common.Address, amounts []*big.Int, nonces []*big.Int, batchNonce *big.Int, signatures [][]byte) (*types.Transaction, error)
```
Comment (dragos-rebegea) [inline clients/ethereum/wrappers/interface.go:21]: maybe make a struct for ExecuteTransfer arguments

File: factory/ethElrondBridgeComponents_test.go [hunk 3]
```diff
@@ -125,6 +119,15 @@ func TestNewEthElrondBridgeComponents(t *testing.T) {
 		assert.Equal(t, errNilStatusStorer, err)
 		assert.Nil(t, components)
 	})
+	t.Run("nil MetricsHolder", func(t *testing.T) {
```
Comment (dragos-rebegea) [inline factory/ethElrondBridgeComponents_test.go:122]: move this before last test ("should work") to mentain the the order as in checkArgsEthereumToElrondBridge.
Also there is not test for empty NetworkAddress(GeneralConfig.Elrond.NetworkAddress)

File: clients/ethereum/wrappers/ethereumChainWrapper.go (hunk 1 above)
Comment (iulianpascalau) [reply clients/ethereum/wrappers/ethereumChainWrapper.go:111]: done

File: clients/ethereum/client.go [hunk 4]
```diff
@@ -395,6 +395,11 @@ func (c *client) getNonce(ctx context.Context, fromAddress common.Address) (int6
 	return int64(nonce), err
 }
 
+// GetTransactionsStatuses will return the transactions statuses from the batch ID
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply clients/ethereum/client.go:398]: done

File: clients/ethereum/wrappers/ethereumChainWrapper_test.go [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+	assert.Nil(t, err)
+	assert.Nil(t, chainID)
+	assert.True(t, handlerCalled)
+	assert.Equal(t, 1, statusHandler.GetIntMetric(core.MetricNumEthClientRequests))
+}
+
+func TestEthClientWrapper_BlockNumber(t *testing.T) {
+	t.Parallel()
+
+	t.Run("block number call errors", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply clients/ethereum/wrappers/ethereumChainWrapper_test.go:156]: done

File: clients/ethereum/wrappers/interface.go (hunk 2 above)
Comment (iulianpascalau) [reply clients/ethereum/wrappers/interface.go:21]: can not do this. The struct that implement this is auto-generated

File: factory/ethElrondBridgeComponents_test.go (hunk 3 above)
Comment (iulianpascalau) [reply factory/ethElrondBridgeComponents_test.go:122]: moved the test & added check

File: factory/ethElrondBridgeComponents_test.go [hunk 6]
```diff
@@ -208,6 +211,16 @@ func TestNewEthElrondBridgeComponents(t *testing.T) {
 		assert.True(t, strings.Contains(err.Error(), "for TimeForBootstrap"))
 		assert.Nil(t, components)
 	})
+	t.Run("should work", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply factory/ethElrondBridgeComponents_test.go:214]: I know. Actually that is pretty hard to test. This always happens in the factories that create instances. I can not make a test for that unless I mess-up with the code

**Review by dragos-rebegea — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]
